### PR TITLE
Update: meguca, quicktemplate, statik, hydron -> Convert to buildGoModule

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
Got meguca working again, so it's no longer marked as broken!
In the process, learned how we have a `buildGoModule` method now, much better than what was used previously!

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Chiiruno** [comment]
cc @Infinisil 
the person ofborg CC'd is on vacation.

**Comment by Chiiruno** [comment]
Ran `nix-shell -p nix-review --run "nix-review wip"`, only a few failures in nodePackages, like usual.

File: nixos/modules/services/web-servers/meguca.nix [hunk 1]
```diff
@@ -28,68 +28,122 @@ in with lib; {
...
+      example = "127.0.0.1:9001";
+      description = "Address to listen on for incoming connections.";
     };
 
-    assumeReverseProxy = mkOption {
+    reverseProxy = mkOption {
       type = types.bool;
       default = false;
-      description = "Assume the server is behind a reverse proxy, when resolving client IPs.";
+
```
Comment (teto) [inline nixos/modules/services/web-servers/meguca.nix]: why the change ? might be best to keep the old name ?

File: nixos/modules/services/web-servers/meguca.nix [hunk 2]
```diff
@@ -28,68 +28,122 @@ in with lib; {
...
+      type = types.path;
+      default = "/run/keys/meguca-server-config-file";
+      example = "/home/okina/meguca/keys/server-config";
+      description = "Meguca server configuration JSON file.";
     };
 
     videoPaths = mkOption {
       type = types.listOf types.path;
-      default = [];
+      default = [ ];
```
Comment (teto) [inline nixos/modules/services/web-servers/meguca.nix]: is that sthg done by a linter ?

File: nixos/modules/services/web-servers/meguca.nix [hunk 3]
```diff
@@ -28,68 +28,122 @@ in with lib; {
...
+        as the administrator postgres user in the psql shell.
+      '';
     };
 
-    postgresArgsFile = mkOption {
-      type = types.path;
-      default = "/run/keys/meguca-postgres-args";
-      example = "/home/okina/meguca/keys/postgres";
-      description = "Postgresql connection arguments file.";
+    cacheSize = mkOption {
```
Comment (teto) [inline nixos/modules/services/web-servers/meguca.nix]: why the move to float ? are we that constrained ?

File: nixos/modules/services/web-servers/meguca.nix [hunk 4]
```diff
@@ -28,68 +28,122 @@ in with lib; {
...
+      '';
     };
 
-    compressTraffic = mkOption {
-      type = types.bool;
-      default = false;
-      description = "Compress all traffic with gzip.";
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1:8000";
```
Comment (teto) [inline nixos/modules/services/web-servers/meguca.nix]: changing the default here looks like a bad idea.

**Review by teto — CHANGES_REQUESTED** [review]
sslCertificate and postgresArgs options disappear, is that ok ? why ? anyway to achieve the same behavior otherwise

**Comment by Chiiruno** [comment]
>sslCertificate and postgresArgs options disappear, is that ok ? why ? anyway to achieve the same behavior otherwise

Yes, meguca doesn't support SSL in postgres, and postgresArgs is deprecated by the new config file, and is impossible to use command arguments now.

File: nixos/modules/services/web-servers/meguca.nix (hunk 4 above)
Comment (ciil) [reply nixos/modules/services/web-servers/meguca.nix]: Seems like it's just been made explicit from internal default before, lgtm.

**Review by ciil — APPROVED** [review]
Just built it and it runs smoothly in a container.

File: nixos/modules/services/web-servers/meguca.nix (hunk 3 above)
Comment (teto) [reply nixos/modules/services/web-servers/meguca.nix]: floats seem indeed fine https://godoc.org/github.com/bakape/meguca/config#ServerConfigs (u64 type would have been fine if range is a concern).

File: nixos/modules/services/web-servers/meguca.nix (hunk 2 above)
Comment (teto) [reply nixos/modules/services/web-servers/meguca.nix]: in which case, it's best not to change.

File: nixos/modules/services/web-servers/meguca.nix [hunk 5]
```diff
@@ -28,68 +28,122 @@ in with lib; {
...
+          "test": {
+            "database": "postgres://meguca:meguca@localhost:5432/meguca_test?sslmode=disable"
+          }
+        }
+      '';
+
+      description = "Meguca server configuration in JSON format.";
+    };
+
+    serverConfigFile = mkOption {
```
Comment (teto) [inline nixos/modules/services/web-servers/meguca.nix]: it is redundant with serverConfig, people can use `serverConfig = builtins.readFile`instead.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build meguca quicktemplate statik hydron

**Comment by Chiiruno** [comment]
I noticed this in the darwin test, despite the test passing, meguca doesn't build due to opencv not being supported.
```
Cannot nix-instantiate `meguca' because:
error: while evaluating the attribute 'buildInputs' of the derivation 'meguca-6.3.0' at /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndndx-vm/pkgs/stdenv/generic/make-derivation.nix:191:11:
while evaluating the attribute 'handled' at /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndndx-vm/pkgs/stdenv/generic/check-meta.nix:251:7:
while evaluating 'handleEvalIssue' at /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndndx-vm/pkgs/stdenv/generic/check-meta.nix:147:38, called from /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndndx-vm/pkgs/stdenv/generic/check-meta.nix:252:14:
Package ‘opencv-2.4.13’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndndx-vm/pkgs/development/libraries/opencv/default.nix:88 is not supported on ‘x86_64-darwin’, refusing to evaluate.
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.
b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.
```
Is this something we should address?

**Comment by Chiiruno** [comment]
Rebase due to #66585.

**Comment by Chiiruno** [comment]
Depends on #68135 
Will update PR with changes once that is merged.

File: nixos/modules/services/web-servers/hydron.nix [hunk 6]
```diff
@@ -80,69 +82,78 @@ in with lib; {
...
+          }
+        '';
       };
-    };
 
-    systemd.services.hydron-fetch = {
-      description = "Import paths into hydron and possibly fetch tags";
+      postgresql = {
+        enable = true;
+        package = pkgs.postgresql_11;
```
Comment (Infinisil) [inline nixos/modules/services/web-servers/hydron.nix]: I don't think this is a good idea. This essentially changes the postgres version for all NixOS services

**Review by infinisil — CHANGES_REQUESTED** [review]
This could really use some splitting up into multiple commits, it's really hard to review like this

File: nixos/modules/services/web-servers/hydron.nix (hunk 6 above)
Comment (Chiiruno) [reply nixos/modules/services/web-servers/hydron.nix]: That's a good point for this package, but there's an issue with that when it comes to the meguca package, which can't use the default of... I think it was 9.6?

**Comment by Chiiruno** [comment]
It was split up into multiple commits before, I wasn't expecting anyone else to review after I said this depends on #68135 , before fixing everything back up.
I just saved some temporary changes into a temporary commit since I had already reset to the master revision.
Anyway, sorry for any inconvenience.

File: nixos/modules/services/web-servers/hydron.nix (hunk 6 above)
Comment (Infinisil) [reply nixos/modules/services/web-servers/hydron.nix]: Maybe have an assertion for the version being above X

**Comment by Chiiruno** [comment]
Ready for review!

**Comment by Chiiruno** [comment]
That should be a bit better.

**Comment by prusnak** [comment]
Please rework your PR. It now has a merge conflict.

**Comment by Chiiruno** [comment]
Outdated, v7 is on it's way.
